src/immutable.ts

```diff
@@ -0,0 +1,36 @@
+import useSWRImmutable from "swr/immutable";
+import { configureBaseQueryHook } from "./query-base.js";
+
+/**
+ * ```ts
+ * const client = createClient();
+ *
+ * const useImmutable = createImmutableHook(client, "<unique-key>");
```

Comment:
Maybe worth a description as to how this hook is different etc since I don't believe useImmutable is part of SWR api